teslajsonpy/controller.py

```diff
@@ -948,78 +1002,163 @@ async def _get_and_process_energysite_data(energysite_id: Text) -> None:
 
             return any(await asyncio.gather(*tasks))
 
-    def get_climate_params(self, car_id):
+    def get_climate_params(self, car_id: Text = None, vin: Text = None) -> Dict:
         """Return cached copy of climate_params for car_id."""
-        vin = self._id_to_vin(car_id)
-        if vin:
+        if car_id and not vin:
+            vin = self._id_to_vin(car_id)
+        if vin and vin in self.__climate:
             return self.__climate[vin]
-        return {}
+        return self.__climate
 
-    def set_climate_params(self, car_id: Text, params: Dict) -> None:
+    def set_climate_params(
+        self, car_id: Text = None, vin: Text = None, params: Dict = None
+    ) -> None:
         """Set climate_params for car_id."""
-        vin = self._id_to_vin(car_id)
+        params = params or {}
+        if car_id and not vin:
+            vin = self._id_to_vin(car_id)
         if vin:
             self.__climate[vin] = params
 
-    def get_charging_params(self, car_id):
+    def is_climate_on(self, car_id: Text = None, vin: Text = None) -> bool:
+        """Return true if climate is on."""
+        if car_id and not vin:
+            vin = self._id_to_vin(car_id)
+        if vin and vin in self.__climate:
+            return self.get_climate_params(vin=vin).get("is_climate_on")
+        return False
+
+    def get_charging_params(self, car_id: Text = None, vin: Text = None) -> Dict:
         """Return cached copy of charging_params for car_id."""
-        vin = self._id_to_vin(car_id)
-        if vin:
+        if car_id and not vin:
+            vin = self._id_to_vin(car_id)
+        if vin and vin in self.__charging:
             return self.__charging[vin]
-        return {}
+        return self.__charging
```

Comment:
Just wanted to confirm this return is intentional and you've considered the use cases. Previously we'd return an empty dict. now it's returning a dict of dicts when the vin doesn't exist. I'm not opposed if you've thought about it.

Reply:
Hi, and thanks for the review.

Yes, this is intentional.  There was some inconsistensies before, where some functions (e.g. `get_updates` and `get_last_update_time` returned the whole dict, while others returned an empty dict, and I wanted to have the same behaviour in all of the functions.
When choosing between different return values, I thought of more (potential) use cases with getting the whole dict, than getting an empty dict.  An other option was to return `None` as that would maybe be the least surprising if no vin or car_id was provided, but again, that would not allow you to get the whole dict, which is useful in cases like `for vin, online in self.get_car_online().items():` (line 960).